x/harvest/keeper/deposit.go

```diff
@@ -42,64 +58,68 @@ func (k Keeper) Deposit(ctx sdk.Context, depositor sdk.AccAddress, amount sdk.Co
 	ctx.EventManager().EmitEvent(
 		sdk.NewEvent(
 			types.EventTypeHarvestDeposit,
-			sdk.NewAttribute(sdk.AttributeKeyAmount, amount.String()),
+			sdk.NewAttribute(sdk.AttributeKeyAmount, coins.String()),
 			sdk.NewAttribute(types.AttributeKeyDepositor, deposit.Depositor.String()),
-			sdk.NewAttribute(types.AttributeKeyDepositDenom, deposit.Amount.Denom),
 		),
 	)
 
 	return nil
 }
 
 // ValidateDeposit validates a deposit
-func (k Keeper) ValidateDeposit(ctx sdk.Context, amount sdk.Coin) error {
+func (k Keeper) ValidateDeposit(ctx sdk.Context, coins sdk.Coins) error {
 	params := k.GetParams(ctx)
-	for _, lps := range params.LiquidityProviderSchedules {
-		if lps.DepositDenom == amount.Denom {
-			return nil
+	for _, depCoin := range coins {
+		found := false
+		for _, lps := range params.LiquidityProviderSchedules {
+			if lps.DepositDenom == depCoin.Denom {
+				found = true
+			}
+		}
+		if !found {
+			return sdkerrors.Wrapf(types.ErrInvalidDepositDenom, "liquidity provider denom %s not found", depCoin.Denom)
 		}
 	}
-	return sdkerrors.Wrapf(types.ErrInvalidDepositDenom, "liquidity provider denom %s not found", amount.Denom)
+
+	return nil
 }
 
 // Withdraw returns some or all of a deposit back to original depositor
-func (k Keeper) Withdraw(ctx sdk.Context, depositor sdk.AccAddress, amount sdk.Coin) error {
-	deposit, found := k.GetDeposit(ctx, depositor, amount.Denom)
+func (k Keeper) Withdraw(ctx sdk.Context, depositor sdk.AccAddress, coins sdk.Coins) error {
+	deposit, found := k.GetDeposit(ctx, depositor)
 	if !found {
-		return sdkerrors.Wrapf(types.ErrDepositNotFound, "no %s deposit found for %s", amount.Denom, depositor)
+		return sdkerrors.Wrapf(types.ErrDepositNotFound, "no deposit found for %s", depositor)
 	}
 
-	if !deposit.Amount.IsGTE(amount) {
-		return sdkerrors.Wrapf(types.ErrInvalidWithdrawAmount, "%s>%s", amount, deposit.Amount)
+	if !deposit.Amount.IsAllGTE(coins) { // TODO test that this works how I think it does
```

Comment:
```suggestion
	if !deposit.Amount.IsAllGTE(coins) {
```

Tested a couple different approaches; this and `!coins.IsAllLTE(deposit.Amount)` both seem to work fine.